tests/test_errors.py

```diff
@@ -455,6 +475,38 @@ def finalize(self, cfg: Any) -> None:
         ),
         id="dict[int,Any]:mistyped_key",
     ),
+    pytest.param(
+        Expected(
+            create=lambda: DictConfig({}, key_type=float),
+            op=lambda cfg: cfg.get("foo"),
+            exception_type=KeyValidationError,
+            msg=dedent(
+                """\
+                Key foo (str) is incompatible with (float)
+                    full_key: foo
+                    object_type=dict"""
+            ),
```

Comment:
compare the message only. other components of the exceptions can be compared directly.
```suggestion
            msg="Key foo (str) is incompatible with (float)"
```